clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -430,6 +434,44 @@ public static String formatAddress(String host, Integer port) {
     }
 
     /**
+     * Formats a byte number as a human readable String ("3.2 MB")
+     * @param bytes some size in bytes
+     * @return
+     */
+    public static String formatBytes(long bytes) {
+        if (bytes < 0) {
+            return "" + bytes;
+        }
+        double asDouble = (double) bytes;
+        int ordinal = (int) Math.floor(Math.log(asDouble) / Math.log(1024.0));
+        double scale = Math.pow(1024.0, ordinal);
+        double scaled = asDouble / scale;
+        String formatted = TWO_DIGIT_FORMAT.format(scaled);
+        switch (ordinal) {
```

Comment:
This is fine. You can also put those all in an array and just index with `min(ordinal, units.length-1)`.